.pre-commit-config.yaml

```diff
@@ -3,36 +3,57 @@ default_language_version:
   python: "3.9"
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
+    rev: "v4.1.0"
     hooks:
       - id: check-yaml
+        exclude: .clang-format
       - id: end-of-file-fixer
       - id: trailing-whitespace
```

Comment:
This is breaking a test that has trailing whitespace in the output. Given that black handles .py files anyway, I can just turn it off for Python. Though not producing trailing whitespace in the first place might be better. ;)

Reply:
Which test? Is it in `test_console.py`?

Reply:
https://github.com/pyodide/pyodide/blob/525d1111b094fc0c551dee9ad5e4f4d4546bdb61/packages/micropip/src/micropip/package.py#L10-L20

There's trailing whitespace there and it's needed for the test.

Reply:
I find `trailing-whitespace` quite useful, particularly for docstrings (are you sure black would handle docstings)?  
Otherwise, for the doctest we could just use `...` to match anything. It's already  enabled.

Reply:
If that specific doctest is the problem, maybe we could simply remove it, since `_format_table` is  a private function used only for `micropip.list()` and we already have a [doctest for micopip.list()](https://pyodide.org/en/stable/usage/api/micropip-api.html#micropip.list).

Reply:
Pretty sure black does strip it, given there's a fmt off/on around this block already, presumably for black not to strip this (otherwise it doesn't touch docstrings).

Can't `_format_table` simply strip trailing whitespace? It's really a bit messy to put it into the output?

Reply:
> Can't `_format_table` simply strip trailing whitespace? It's really a bit messy to put it into the output?

Yeah, that could be better. I made a PR for that #2181 :)